src/components/Referrals/ReferralRewardCard/ReferralRewardCard.tsx

```diff
@@ -1,68 +1,92 @@
-import React from 'react'
-import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined'
+import React, { useState } from 'react'
+import CheckRoundedIcon from '@mui/icons-material/CheckRounded'
+import LockRoundedIcon from '@mui/icons-material/LockRounded'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
-import { Tooltip } from 'decentraland-ui2'
-import type { ReferralTier } from '../../../modules/referrals/types'
+import ClaimRewardModal from './ClaimRewardModal'
 import {
   REFERRAL_REWARD_CARD_TEST_ID,
-  REFERRAL_REWARD_TIER_TEST_ID,
   REFERRAL_REWARD_IMAGE_TEST_ID,
   REFERRAL_REWARD_RARITY_TEST_ID,
-  REFERRAL_REWARD_SWAG_CONTAINER_TEST_ID,
-  REFERRAL_REWARD_DESCRIPTION_TEST_ID,
-  REFERRAL_REWARD_CLAIM_BUTTON_TEST_ID
+  REFERRAL_REWARD_CLAIM_BUTTON_TEST_ID,
+  REFERRAL_REWARD_CHECK_ICON_TEST_ID,
+  REFERRAL_REWARD_LOCK_ICON_TEST_ID
 } from './constants'
 import {
   GradientBorder,
   CardContainer,
   ClaimButton,
-  Description,
   RewardRarity,
   RewardImage,
-  RewardLabel,
   RewardImageContainer,
-  SwagRarityContainer
+  ReferralRewardCardContainer,
+  ReferralRewardCardHeader,
+  ReferralRewardCardTitle,
+  HeaderContainer,
+  CompletedIcon
 } from './ReferralRewardCard.styled'
+import { ReferralRewardCardProps } from './ReferralRewardCard.types'
 
 // eslint-disable-next-line import/no-named-as-default-member
-const ReferralRewardCard = React.memo((props: ReferralTier) => {
-  const { invitesAccepted, rarity, completed, image, claim } = props
+const ReferralRewardCard = React.memo((props: ReferralRewardCardProps) => {
+  const { invitesAccepted, rarity, completed, image, onSetReferralEmail } = props
+  const [isModalOpen, setIsModalOpen] = useState(false)
+
+  const handleClaimClick = () => {
```

Comment:
Would you mind wrapping these functions with `useCallback` to prevent re-renders?